build_runner/lib/src/generate/build_result.dart

```diff
@@ -56,6 +64,16 @@ enum BuildStatus {
   failure,
 }
 
+/// The type of failure
+///
+/// Values correspond to proposed exitCode.
+class FailureType {
+  static const general = const FailureType(1);
+  static const cantCreate = const FailureType(73);
+  final int value;
+  const FailureType(this.value);
```

Comment:
Make this private? Keep the set of exit codes restricted to the ones provided here?